src/Metalama.Patterns.Observability/Implementation/ClassicStrategy/ClassicObservableExpression.cs

```diff
@@ -97,7 +97,7 @@ public ClassicObservableExpression(
                 false => InpcBaseHandling.NotApplicable,
 
                 true when this.IsRoot =>
-                    fieldOrProperty.DeclaringType == builder.CurrentType
+                    fieldOrProperty.DeclaringType.Equals( builder.CurrentType )
```

Comment:
What is the impact of these changes? If this is a bug, there should be tests.

Reply:
I don't know if this is actually a bug, but `Equals` should be safer.